drivers/SmartThings/zigbee-switch/src/hanssem/init.lua

```diff
@@ -0,0 +1,148 @@
+local stDevice = require "st.device"
+local capabilities = require "st.capabilities"
+local zigbeeDriver = require "st.zigbee"
+local zclClusters = require "st.zigbee.zcl.clusters"
+local zclGlobalCommands = require "st.zigbee.zcl.global_commands"
+local dataTypes = require "st.zigbee.data_types"
+local Status = require "st.zigbee.generated.types.ZclStatus"
+
+local OnOff = zclClusters.OnOff
+local Basic = zclClusters.Basic
+
+local utilities = require "hanssem/utilities"
+
+local FINGERPRINTS = {
+  { mfr = "Winners", model = "HS2-P1Z3-1" },
+  { mfr = "Winners", model = "HS2-P1Z3-2" },
+  { mfr = "Winners", model = "HS2-P1Z3-3" },
+  { mfr = "Winners", model = "HS2-P2Z3-4" },
+  { mfr = "Winners", model = "HS2-P2Z3-5" },
+  { mfr = "Winners", model = "HS2-P2Z3-6" }
+}
+
+local function can_handle_hanssem_switch(opts, driver, device, ...)
+  for _, fingerprint in ipairs(FINGERPRINTS) do
+    if device:get_manufacturer() == fingerprint.mfr and device:get_model() == fingerprint.model then
+      return true
+    end
+  end
+  return false
+end
+
+-- Handler Send Command 
+function handlers_sendOn(driver, device)
+  local isParent = device.network_type ~= stDevice.NETWORK_TYPE_CHILD
+  local parent = isParent and device or device:get_parent_device()
+
+  local index
+
+  if isParent then 
+    index = device.fingerprinted_endpoint_id
+  else
+    index = tonumber(device.parent_assigned_child_key)
+  end
+
+  parent:send(OnOff.server.commands.On(parent):to_endpoint(index))
+end
+
+function handlers_sendOff(driver, device)
+  local isParent = device.network_type ~= stDevice.NETWORK_TYPE_CHILD
+  local parent = isParent and device or device:get_parent_device()
+
+  local index
+
+  if isParent then 
+    index = device.fingerprinted_endpoint_id
+  else
+    index = tonumber(device.parent_assigned_child_key)
+  end
```

Comment:
all of this mapping should be handled by your `find_child` functions.

You may want to check the documentation here: https://developer.smartthings.com/docs/edge-device-drivers/zigbee/device.html#parent-child-devices

Reply:
I modified functions according to the guide.

Reply:
merge init.lua and utilities.lua into init.lua.

Reply:
I modified fingerprint.lua and init.lua by changing the model name